clippy_utils/src/lib.rs

```diff
@@ -1786,6 +1786,46 @@ pub fn is_some_ctor(cx: &LateContext<'_>, res: Res) -> bool {
     false
 }
 
+pub fn derefs_to_slice<'tcx>(
+    cx: &LateContext<'tcx>,
+    expr: &'tcx hir::Expr<'tcx>,
+    ty: Ty<'tcx>,
+) -> Option<&'tcx hir::Expr<'tcx>> {
```

Comment:
Could you add documentation that explains this function?
Also, I'm not sure it's necessary to move this under `clippy_utils`.

Reply:
Now [`lint_iter_cloned_collect`](https://github.com/rust-lang/rust-clippy/blob/d5223be2e30a9d116b839bda418bec99d2949a68/clippy_lints/src/methods/mod.rs#L2424), [`lint_iter_nth`](https://github.com/rust-lang/rust-clippy/blob/d5223be2e30a9d116b839bda418bec99d2949a68/clippy_lints/src/methods/mod.rs#L2603), [`lint_iter_next`](https://github.com/rust-lang/rust-clippy/blob/d5223be2e30a9d116b839bda418bec99d2949a68/clippy_lints/src/methods/mod.rs#L2546) and so on use `derefs_to_slice` defined in clippy_lints/src/methods/mod.rs. And I want to use it in iter_count.rs. That's why I  moved it under `clippy_utils`.
You mean that I should redefine `derefs_to_slice` in `iter_count.rs` or use the other function?

Reply:
I think this function is not so general that can be used from the outside of `methods`.  You can use it in `iter_count.rs` without moving it to `clippy_utils`, right? Or you can add another module `methods/utils.rs` and move it under the module.

Reply:
Thank you. I moved `derefs_to_slice` to `methods/mod.rs` and made it visible within the crate in order to use it in `iter_count.rs`.

Reply:
You don't need to add `pub(crate)` to use it inside `methods` module.

Reply:
Oh, I was mistaken thinking that `derefs_to_slice` and `iter_count.rs` are in different modules.